### PR TITLE
Address edge cases in docstring parsing

### PR DESCRIPTION
**Pull request by BenWilson2**
This is intended to fix several edge cases with arg and return naming conventions (added tests to cover these use cases)

### CODE REVIEW

File: src/ucai/core/utils/docstring_utils.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+        raise ValueError(
+            "Docstring is empty. Please provide a docstring with a function description."
+        )
+
+    description_lines = []
+    parsed_params = {}
+    returns = None
+    current_param = None
+    param_description_lines = []
+    return_description_lines = []
```
Comment (harupy) [inline src/ucai/core/utils/docstring_utils.py]: I'd recommend adding type hints if we do "initialize-then-mutate".

Comment (harupy) [reply src/ucai/core/utils/docstring_utils.py]:
```suggestion
    description_lines: list[str] = []
    parsed_params: dict[str, str] = {}
    returns: str | None = None
    current_param str | None = None
    param_description_lines: list[str] = []
    return_description_lines: list[str] = []
```

File: tests/core/test_docstring_utils.py [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+        int: The input value incremented by one.
+    """
+    with pytest.raises(ValueError, match="Function description is missing"):
+        parse_docstring(docstring)
+
+
+def test_parse_docstring_single_line_single_param():
+    """
+    Test parsing a simple docstring with a single-line description and one parameter.
+    """
```
Comment (harupy) [inline tests/core/test_docstring_utils.py]: Do we need this docstring? I think the function name is descriptive enough.

File: tests/core/test_docstring_utils.py [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+
+from ucai.core.utils.docstring_utils import DocstringInfo, parse_docstring
+
+
+def test_parse_docstring_empty():
+    """
+    Test that an empty docstring raises a ValueError.
+    """
+    with pytest.raises(ValueError, match="Docstring is empty"):
+        parse_docstring("")
```
Comment (harupy) [inline tests/core/test_docstring_utils.py]:
```suggestion
def test_parse_docstring_empty_raises_value_error():
    with pytest.raises(ValueError, match="Docstring is empty"):
        parse_docstring("")
```

File: tests/core/test_docstring_utils.py (hunk 2 above)
Comment (harupy) [reply tests/core/test_docstring_utils.py]: docstring might make sense for a complex test case, but I don't think we need docstring for simple test cases.

File: tests/core/test_docstring_utils.py [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+def test_parse_docstring_google_style():
+    """
+    Test parsing a Google-style docstring.
+    """
+    docstring = """
+    Calculate the sum of two numbers.
+
+    Args:
+        a (int): The first number.
+        b (int): The second number.
```
Comment (harupy) [inline tests/core/test_docstring_utils.py:100]:
```suggestion
        a (int): The first number.

        b (int): The second number.
```

what would happen if there is a blank line between a and b? Is this case handled?

File: tests/core/test_docstring_utils.py [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+    Process data.
+
+    Args:
+        data:
+        config:
+    Returns:
+        bool: Success status.
+    """
+    expected = DocstringInfo(
+        description="Process data.", params={}, returns="bool: Success status."
```
Comment (harupy) [inline tests/core/test_docstring_utils.py]: shouldn't params be `{"a": "", "b": ""}` or `{"a": None, "b": None}`?

File: tests/core/test_docstring_utils.py [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+    assert result == expected
+
+
+def test_parse_docstring_no_params():
+    """
+    Test parsing a docstring with no parameters.
+    """
+    docstring = """
+    Get the current timestamp.
+
```
Comment (serena-ruan) [inline tests/core/test_docstring_utils.py:81]: Can we also add test for
```
Args:
Returns:...
```

File: tests/core/test_docstring_utils.py (hunk 2 above)
Comment (BenWilson2) [reply tests/core/test_docstring_utils.py]: Removed them all. The titles are explanatory enough

File: tests/core/test_docstring_utils.py (hunk 4 above)
Comment (BenWilson2) [reply tests/core/test_docstring_utils.py:100]: this worked, but added tests that cover this

File: tests/core/test_docstring_utils.py (hunk 5 above)
Comment (BenWilson2) [reply tests/core/test_docstring_utils.py]: yep, adjusted accordingly

File: tests/core/test_docstring_utils.py (hunk 6 above)
Comment (BenWilson2) [reply tests/core/test_docstring_utils.py:81]: yep, this correctly parses now (added tests too)

File: src/ucai/core/utils/docstring_utils.py [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+                current_section = section
+                content = stripped_line[len(title) :].strip()
+                if content:
+                    sections[current_section].append(content)
+                matched_section = True
+                break
+        if matched_section:
+            continue
+        sections[current_section].append(line)
+    return sections
```
Comment (harupy) [inline src/ucai/core/utils/docstring_utils.py:74]: Not a blocker, but I'd return a dataclass.

**Review by harupy — APPROVED** [review]
LGTM

**Review by serena-ruan — APPROVED** [review]
Thanks for the fix! Let's port this over to unitycatalog repo :D
